02_knight_and_princess/tevfik_aksoy/main.cpp

```diff
@@ -0,0 +1,145 @@
+#include <fstream>
+#include <iostream>
+#include <set>
+#include <string>
+#include <tuple>
+#include <vector>
+
+using std::cout;
+using std::endl;
+using std::get;
+using std::to_string;
+using std::vector;
+
+using NodeId = int;
+using EdgeId = int;
+
+// Tuple of <int,int> for realization of vector of vectors of <int,int>
+using dest_edge_id = std::tuple<int, int>;
+void generateVertices(vector<vector<dest_edge_id>>& graphVec,
+                      vector<std::pair<int, int>>& edges_src_dest);
+// Structure of nodes(vertices)
+struct Node {
+ public:
+  NodeId id = 0;
+  explicit Node(NodeId _id = 0) : id(_id) {}
+
+  std::string to_JSON(const vector<int>& edge_ids) const {
+    std::string str;
+
+    str += "\t{ \"id\": " + to_string(id) + ", \"edge_ids\": [";
+    for (int i = 0; i < edge_ids.size(); i++) {
+      str += to_string(edge_ids[i]);
+      if (i + 1 != edge_ids.size())
+        str += ", ";
+    }
+    return str;
+  }
+};
+
+// Structure of the edge including the source and destination nodes in it
+struct Edge {
+ public:
+  EdgeId id = 0;
+  Node src_node;
+  Node dest_node;
+  Edge(int src_id, int dest_id, int _id)
+      : id(_id), src_node(src_id), dest_node(dest_id) {}
+
+  std::string to_JSON() const {
+    std::string str;
+    str += "\t{ \"id\": " + to_string(id) + ", \"vertex_ids\": [" +
+           to_string(src_node.id) + ", " + to_string(dest_node.id) + "] }";
+    return str;
+  }
+};
+
+class Graph {
+ private:
+  vector<std::pair<int, int>>
+      edges_src_dest_;  // vector of pairs to keep data of source and

```

Comment:
Тоже самое, наверное ты хотел использовать `NodeId` здесь?
```suggestion
  vector<std::pair<NodeId, NodeId>>
      edges_src_dest_;  // vector of pairs to keep data of source and
```

Reply:
Исправил